controllers/front/OrderController.php

```diff
@@ -237,6 +237,26 @@ public function displayAjaxselectDeliveryOption()
         ]));
     }
 
+    public function displayAjaxCheckCartStillOrderable(): void
+    {
+        $responseData = [
+            'errors' => false,
+            'cartUrl' => '',
+        ];
+
+        // TODO - move to separate single function in Cart and also check if discounts and rules are valid
+        if ($this->context->cart->isAllProductsInStock() !== true ||
+            $this->context->cart->checkAllProductsAreStillAvailableInThisState() !== true ||
+            $this->context->cart->checkAllProductsHaveMinimalQuantities() !== true) {
+            $responseData['errors'] = true;
+            $responseData['cartUrl'] = $this->context->link->getPageLink('cart', null, null, ['action' => 'show']);
+        }
+
+        ob_end_clean();
```

Comment:
I think you should remove this ;)
```suggestion
```

Reply:
This line is added before each ajax returns to avoid notice and warning that can broke json response.

I think we should keep it in order to avoid broken checkout tunnel in case of a notice or a warning is printed before json, because we are on the payment step and this is critical to have a response can be parsed by current javascript implementation.

The `ob_start` is on FrontControllerCore : https://github.com/PrestaShop/PrestaShop/blob/1.7.8.0/classes/controller/FrontController.php#L329

And many controllers have this line before returns json, for example :
https://github.com/PrestaShop/PrestaShop/blob/1.7.8.0/controllers/front/AddressController.php#L178
https://github.com/PrestaShop/PrestaShop/blob/1.7.8.0/controllers/front/AttachmentController.php#L38
https://github.com/PrestaShop/PrestaShop/blob/1.7.8.0/controllers/front/CartController.php#L174
https://github.com/PrestaShop/PrestaShop/blob/1.7.8.0/controllers/front/CartController.php#L225
https://github.com/PrestaShop/PrestaShop/blob/1.7.8.0/controllers/front/GetFileController.php#L300
https://github.com/PrestaShop/PrestaShop/blob/1.7.8.0/controllers/front/OrderController.php#L230
https://github.com/PrestaShop/PrestaShop/blob/1.7.8.0/controllers/front/OrderController.php#L320
https://github.com/PrestaShop/PrestaShop/blob/1.7.8.0/controllers/front/ProductController.php#L423
https://github.com/PrestaShop/PrestaShop/blob/1.7.8.0/controllers/front/ProductController.php#L450

Reply:
The same goes for https://github.com/PrestaShop/PrestaShop/pull/25353.

@PierreRambaud @atomiix @Matt75 Guys pls agree on one thing and tell me the result 😄

Reply:
I kinda agree with @PierreRambaud, we shouldn't be hiding errors like that. If you don't want warning or notice, ou should configure your php.ini IMO